src/Components/TaskScheduler.php

```diff
@@ -17,24 +19,95 @@ class TaskScheduler
     public static function schedule(SchedulerTask $task)
     {
         if (empty($task->payload)) {
-            $data = "";
+            $data = '';
         } else {
             $data = json_encode(json_decode($task->payload));
         }
+
         $verb = VerbsMask::toString($task->verb_mask);
         $serviceName = \ServiceManager::getServiceById($task->service_id)->getName();
         $component = $task->component;
         $commandOptions = '--format=JSON --verb=' . $verb . ' --service=' . $serviceName . ' --resource=' . $component;
 
         // Use the scheduler to schedule the task at its desired frequency in minutes
         if ($task->is_active) {
+
+            $dirPath = storage_path() . DIRECTORY_SEPARATOR . 'logs' . DIRECTORY_SEPARATOR . 'tmp';
+
+            if (!file_exists($dirPath)) mkdir($dirPath, 0777, true);
+
+            if ($logFile = self::createTaskLogFile($dirPath)) {
+                $logFilePath = stream_get_meta_data($logFile)['uri'];
+            } else {
+                $logFilePath = $dirPath . DIRECTORY_SEPARATOR . uniqid('schedule', true) . '.log';;
+            }
+
             app(Schedule::class)
                 ->command('df:request \'' . $data . '\' ' . $commandOptions)
                 ->cron('*/' . $task->frequency . ' * * * *')
                 ->withoutOverlapping()
-//                ->appendOutputTo(storage_path() . '/scheduled-tasks-responses.log')
-                ->onFailure(function () use ($task) {
+                ->sendOutputTo($logFilePath)
+                ->onSuccess(function () use ($task, $logFilePath) {
+                    if (TaskLog::whereTaskId($task->id)->exists()) {
+                        TaskLog::whereTaskId($task->id)->first()->delete();
+                    }
+                    if (file_exists($logFilePath)) unlink($logFilePath);
+                })
+                ->onFailure(function () use ($task, $logFilePath) {
+                    try {
```

Comment:
Creating big anonymous methods is a bad practice. It is better to break down this method into at least two small

Reply:
I don't know how I missed that. Refactored to several smaller functions